myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExecutorCommandLineGenerator.java

```diff
@@ -19,11 +19,148 @@
+    pull the config from the resource manager and put them in the yarnConfiguration dir.  This is also why we need
+    frameworkSuperUser. This will be refactored after Mesos-1790 is resolved.
+   */
+
+    //TODO(DarinJ) support other compression, as this is a temp fix for Mesos 1760 may not get to it.
+    if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
+      //Extract tarball keeping permissions, necessary to keep HADOOP_HOME/bin/container-executor suidbit set.
+      //If SudoUser not enable LinuxExecutor will not work
+      appendSudo(cmdLine);
+      cmdLine.append("tar -zxpf ").append(getFileName(myriadExecutorConfiguration.getNodeManagerUri().get()));
+      cmdLine.append(" && ");
+      //Place the hadoop config where in the HADOOP_CONF_DIR where it will be read by the NodeManager
+      //The url for the resource manager config is: http(s)://hostname:port/yarnConfiguration so fetcher.cpp downloads the
+      //config file to yarnConfiguration, It's an xml file with the parameters of yarn-site.xml, core-site.xml and hdfs.xml.
+      if (!myriadExecutorConfiguration.getConfigUri().isPresent()) {
+        appendSudo(cmdLine);
+        cmdLine.append(" cp yarnConfiguration ");
+        cmdLine.append(myriadConfiguration.getYarnEnvironment().get("YARN_HOME"));
+        cmdLine.append("/etc/hadoop/yarn-site.xml && ");
+      }
+    }
+  }
+
+  protected void addJavaOpt(StringBuilder opts, String propertyName, String propertyValue) {
+    String envOpt = String.format(PROPERTY_FORMAT, propertyName, propertyValue);
+    opts.append(envOpt);
+  }
+
+  protected void appendSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo ");
+    }
+  }
+
+  protected void appendUserSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo -E -u ");
+      cmdLine.append(myriadConfiguration.getFrameworkUser().get());
+      cmdLine.append(" -H ");
+    }
+  }
+
+  public String getConfigurationUrl() {
+    String httpPolicy = yarnConfiguration.get(TaskFactory.YARN_HTTP_POLICY);
+    String address = StringUtils.EMPTY;
+    if (httpPolicy != null && httpPolicy.equals(TaskFactory.YARN_HTTP_POLICY_HTTPS_ONLY)) {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+      }
+    } else {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+      }
+    }
+
+    return "http://" + address + "/yarnConfiguration";
+  }
+
+  private static String getFileName(String uri) {
```

Comment:
Recommend throwing an unchecked exception such as IllegalArgumentException if the filename is invalid. I may be missing something, but it seems like we don't want the command line generation to continue for invalid file names